Update: Performance Docs (#1801)
getsentry/sentry-docs: src/collections/_documentation/performance-monitoring/performance/event-detail.md

## Span View

The span view is a split view where the left-hand side shows the transaction’s span tree, and the right-hand side represents each span as a colored rectangle. Within the tree view, Sentry identifies spans by their `op` and `description` values. If a span doesn’t have a description, Sentry uses the span’s id as a fallback. The first span listed is always the transaction’s root span, from which all other spans in the transaction descend.

[{% asset performance/span-details.png alt="Span detail view shows the span id, trace id, parent span id, and other data such as tags." %}]({% asset performance/span-details.png @path %})

> To find these views, you can either go through the [Transaction Summary](/performance-monitoring/performance/transaction-summary) or [Query Builder](/performance-monitoring/discover-queries/query-builder/). Event IDs will be linked to open the corresponding Event Detail.

Author: @doralchan I updated lines 29 and 33 to align with this paragraph. Will you double check the accuracy?

[{% asset performance/span-details.png alt="Span detail view shows the span id, trace id, parent span id, and other data such as tags." %}]({% asset performance/span-details.png @path %})

> You can view span data using either of these methods:

Reviewer: Let's rephrase this - it's as if you're describing how to get to a different view when you are just explaining how you get to _this_ view. For example: 

`To find these views, you can either go through the [Performance Summary] or [Query Builder]. Event IDs will be linked to open the corresponding Event Detail. `

> 1. Click the "Related Transaction" button from the Issue Details page to open the Transaction summary. This will maintain the context of the current error event.

Reviewer 2: @MimiDumpling this button has now been moved to the "trace" context panel as "View Summary": 
<img width="1329" alt="Screen Shot 2020-06-26 at 5 53 53 PM" src="https://user-images.githubusercontent.com/139499/85904295-511d9900-b7d6-11ea-8fc1-faf6908a76db.png">

2. Select the [Performance Homepage](/performance-monitoring/performance/index), then click the affected transaction to display the trace data.

> _Note_: Users on the Team or Business plans can also view a list of transaction events by clicking on the "Transactions" pre-built query in [Discover](/performance-monitoring/discover-queries/index) or by performing a search with the event.type:transaction condition the Discover Query Builder view.

Reviewer 2: ```suggestion
_Note_: Users on the Team or Business plans can also view a list of transaction events by clicking on the "Transactions" pre-built query in [Discover](/performance-monitoring/discover-queries/index) or by performing a search with the `event.type:transaction` condition the Discover Query Builder view.
```

    content=__alert_content
    level="info"
%}

> **Grouping Transactions**

Author: Good call. I'll move it to the setup page.

When Sentry captures transactions, they are assigned a transaction name. This name is generally auto-generated by the Sentry SDK based on the framework integrations you are using. If you can't leverage the automatic transaction generation (or want to customize how transaction names are generated) you can use a global event processor that is registered when you initialize the SDK with your configuration.

An example of doing this in a node.js application:

```javascript
import { addGlobalEventProcessor } from '@sentry/node';

addGlobalEventProcess(event => {
  // if event is a transaction event
  if (event.type === "transaction") {
	event.transaction = sanitizeTransactionName(event.transaction);
  }
  return event;
});
```

For browser JavaScript applications using the `Tracing` integration, the `beforeNavigate` option  can be used to better group navigation/pageload transactions together based on URL.

```javascript
import * as Sentry from '@sentry/browser';
import { Integrations as ApmIntegrations } from '@sentry/apm';

Sentry.init({
	// ...
  integrations: [
    new ApmIntegrations.Tracing({

>       beforeNavigate: (location) => {        
>         // You could use your UI's routing library to find the matching
> 		// route template here. We don't have one right now, so do some basic
> 		// parameter replacements.

Reviewer 3: ```suggestion
      beforeNavigate: (location) => {        
        // You could use your UI's routing library to find the matching
        // route template here. We don't have one right now, so do some basic
        // parameter replacements.
```

Looks like you might have mixed tabs/spaces here or the indentation is just inconsistent.

        // You could use your UI's routing library to find the matching
		    // route template here. We don't have one right now, so do some basic
		    // parameter replacements.
        return location.pathname
        .replace(/\d+/g, '<digits>')
        .replace(/[a-f0-9]{32}/g, '<hash>');
      },
    }),
  ],
});
```

**Adding Query Information and Parameters to Spans**

Currently, every tag has a maximum character limit of 200 characters. Tags over the 200 character limit will become truncated, losing potentially important information. To retain this data, you can split data over several tags instead.

For example, a 200+ character tagged database query:

`SELECT first_column, second_column, third_columns, fourth_column, fifth_column, sixth_column FROM this_is_a_long_table_name WHERE first_column=some_value AND second_column=some_other_value AND third_column=yet_another_value`

The 200+ character query above will become truncated to:

`SELECT first_column, second_column, third_columns, fourth_column, fifth_column, sixth_column FROM this_is_a_long_table_name WHERE first_column=some_value AND second_column=some_other_value AND third_c`

> Instead, using `span.set_tag` you could split the details of this query over several tags. This could be done over `columns`, `tables`, `conditions`, in this case, resulting in three different tags:
> 
> ```python
> import sentry_sdk
> 
> ...
> 
> with sentry_sdk.start_span(op="db", transaction="query api") as span:
>     span.set_tag("columns", columns)
>     span.set_tag("tables", tables)
>     span.set_tag("conditions", conditions)
>     span.set_tag("query_format", query_format)
>     query_format = "SELECT {columns} FROM {tables} WHERE {conditions}"
>     query = query_format.format(
>         columns=columns,
>         tables=tables,
>         conditions=conditions,
>     )
>     ...
> ```

Reviewer 4: ```suggestion
Instead, using `span.setTag` you could split the details of this request over several tags. This could be done over `baseUrl`, `endpoint`, `parameters`, in this case, resulting in three different tags:

```javascript
const span = transaction.startChild({
  op: "request",
  description: "setup form"
});

span.setTag("baseUrl", baseUrl);
span.setTag("endpoint", endpoint);
span.setTag("parameters", parameters);
http.get(`${base_url}/${endpoint}/`, data=parameters);
...
```
Python Version:

```suggestion
Instead, using `span.set_tag` you could split the details of this query over several tags. This could be done over `base_url`, `endpoint`, `parameters`, in this case, resulting in three different tags:

```python
import sentry_sdk

...

with sentry_sdk.start_span(op="request", transaction="setup form") as span:
    span.set_tag("base_url", base_url)
    span.set_tag("endpoint", endpoint)
    span.set_tag("parameters", parameters)
    make_request(
        "{base_url}/{endpoint}/".format(
            base_url=base_url,
            endpoint=endpoint,
        ),
        data=parameters
    )
    ...
```
```

> columns
> : `first_column, second_column, third_columns, fourth_column, fifth_column, sixth_column`
> 
> tables
> : `this_is_a_long_table_name`
> 
> conditions
> : `first_column=some_value AND second_column=some_other_value AND third_column=yet_another_value`

Reviewer 4: ```suggestion
baseUrl
: `https://empowerplant.io`

endpoint
: `api/0/projects/ep/setup_form`

parameters
: ```{
"user_id": 314159265358979323846264338327,
"tracking_id": "EasyAsABC123OrSimpleAsDoReMi",
"product_name": PlantToHumanTranslator,
"product_id": 161803398874989484820458683436563811772030917980576,
}```
```

        return location.pathname
        .replace(/\d+/g, '<digits>')
        .replace(/[a-f0-9]{32}/g, '<hash>');
      },
    }),
  ],
});
```

**Adding Query Information and Parameters to Spans**

Currently, every tag has a maximum character limit of 200 characters. Tags over the 200 character limit will become truncated, losing potentially important information. To retain this data, you can split data over several tags instead.

For example, a 200+ character tagged database query:

`SELECT first_column, second_column, third_columns, fourth_column, fifth_column, sixth_column FROM this_is_a_long_table_name WHERE first_column=some_value AND second_column=some_other_value AND third_column=yet_another_value`

The 200+ character query above will become truncated to:

`SELECT first_column, second_column, third_columns, fourth_column, fifth_column, sixth_column FROM this_is_a_long_table_name WHERE first_column=some_value AND second_column=some_other_value AND third_c`

Instead, using `span.set_tag` you could split the details of this query over several tags. This could be done over `columns`, `tables`, `conditions`, in this case, resulting in three different tags:

> ```python

Reviewer 2: @MimiDumpling do you know if it's a good idea if we should provide examples for each language/framework that has Performance support? For example, Node.js, Python, and browser JavaScript.

Reviewer 4: Actually looking at the other examples should we do everything in JavaScript first, and add more languages over time? (Or now I guess)

I could flip the example query to a long url, and split by URL params instead so that its more applicable?

Author: Ideally, we'd have an example for every supported lang, like Alberto suggested. If you can get me the content, I can add a dropdown for each language.
